server.js

```diff
@@ -730,167 +729,136 @@ app.post('/submit-match',  asyncMiddleware( async (req, res, next) => {
-                                        + JSON.stringify(pending_matches[pending_matches.length - 1]));
-                                    pending_matches.unshift( pending_matches.pop() );
-                                }
-
-                                // Now, if we are playing vs best_network_hash and we have SPRT pass, promote new network.
-                                //
-                                // Actually if we do async functions in here, we might pop wrong stuff off the queue. Better to just check
-                                // at the end and not try to reduce database lookups?
-                                //
-                                // Ok new problem, during the async stuff later more requests come in and so network2=null matches
-                                // don't face right opponent. Lets do a sync copy if this was in fact a new best network
-                                // situation for the current active match in queue.
-                                //
-                                // added !is_test flag
-                                //
-                                if (req.body.loserhash == best_network_hash && !pending_matches[pending_matches.length - 1].is_test) {
-                                    new_best_network_flag = true;
-
-                                    fs.copyFileSync(__dirname + '/network/' + req.body.winnerhash + '.gz', __dirname + '/network/best-network.gz');
-                                    console.log("New best network copied from (fast check): " + __dirname + '/network/' + req.body.winnerhash + '.gz');
-                                }
-                            }
-                        } else {
-                            // network1 was the winner but it was no longer at the end of the pending_match queue.
-                            //
-                        }
-                    }
-                }
+        // Not inserted, we got duplicate sgfhash, abort!
+        if (!dbres.upsertedId) {
+            console.error("Upload match with duplicate sgf");
+            return res.status(400).send("Upload match with duplicate sgf");
+        }
+
+        console.log(req.ip + " (" + req.headers['x-real-ip'] + ") " + " uploaded match " + sgfhash);
+        res.send("Match data " + sgfhash + " stored in database\n");
+    } catch (err) {
+        console.error("Error processing sgffile in /submit-match: " + err);
+        return res.status(400).send("Error with sgf");
+    }
+
+    // prepare $inc 
+    var $inc = { game_count: 1 };
+    if (match.network1 == req.body.winnerhash)
+        $inc.network1_wins = 1;
+    else
+        $inc.network1_losses = 1;
+
+    // save to database using $inc and get modified document
+    match = (await db.collection("matches").findOneAndUpdate(
+        { _id: match._id },
+        { $inc: $inc },
+        { returnOriginal: false }  // return modified document 
```

Comment:
The latest 3.6 documentation seems to have a `returnNewDocument` but not a `returnOriginal`? And since `findOneAndUpdate` was added in 3.2, it looks like it was always `returnNewDocument` ??

https://docs.mongodb.com/v3.2/reference/method/db.collection.findOneAndUpdate/
https://docs.mongodb.com/manual/reference/method/db.collection.findOneAndUpdate/ (3.6)

Looks like this change requires updating package.json.